tracer/src/Datadog.Trace/Agent/IApiRequest.cs

```diff
@@ -12,6 +12,8 @@ internal interface IApiRequest
     {
         void AddHeader(string name, string value);
 
+        Task<IApiResponse> GetAsync();
```

Comment:
Cool, I'm doing the same in https://github.com/DataDog/dd-trace-dotnet/pull/2961/files . I'll wait for this to be merged first then. I can wait.

Reply:
Oh, I hope you will not have a lot of conflicts, after this is merged 😅